inc/Data/ThirdPartyDataFormatter.php

```diff
@@ -181,6 +197,14 @@ public static function formatUrl(string $url, array $params, array $args, array
 
         if ($params && $args) {
             $queryArgs = self::intersectArgs($args, $params);
+            if ($optionalParams) {
+                $optionalArgs = self::intersectArgs($optionalParams, $params);
+                foreach ($optionalArgs as $k => $v) {
+                    if (!isset($queryArgs[$k])) {
```

Comment:
I assume avoiding overwriting existing values is to match the JS behavior?

Reply:
Partially yes. Mostly this uses a `foreach` to have the query parameters sorted based on `params`. I initially used `array_merge($optionalArgs, $queryArgs)`, but that will lead for values from `optionalParams` to come first.

Not a real problem I assume, but it leads to different URLs than the reference snippets for e.g. Google Analytics recommend, so I thought it's worth keeping it consistent with those.